Mobile v2.2 Changelog (#6245)
mattermost/docs: source/deploy/mobile-app-changelog.md

- Server Versions Supported: Server v7.1.0+ is required. Self-Signed SSL Certificates are not supported unless the user installs the CA certificate on their device.

### Compatibility
 - **Upgrade to server version v7.1.0 or later is required.** Support for server [Extended Support Release](https://docs.mattermost.com/upgrade/extended-support-release.html) (ESR) 6.3.0 has ended and upgrading to server ESR v7.1.0 or later is required. As we innovate and offer newer versions of our mobile apps, we maintain backwards compatibility only with supported server versions. Users who upgrade to the newest mobile apps while being connected to an unsupported server version can be exposed to compatibility issues, which can cause crashes or severe bugs that break core functionality of the app.
 - Android operating system 7+ [is required by Google](https://android-developers.googleblog.com/2017/12/improving-app-security-and-performance.html).	
 - iPhone 5s devices and later with iOS 12.1+ is required.

### Improvements
 - Added the ability to set the mobile notifications preferences per channel.
 - Calls: Added slash commands to start/stop call recordings (``/call recording [start|stop]``).

>  - Calls: Implemented a glare free negotiation.

Reviewer: Curious to know more about what this means

Reviewer 2: It's a problem with Webrtc where if two people try to simultaneously connect to each other, the negotiation fails (they can't fully connect).

Reviewer: Thanks, @cpoile! Do we expect end users to be familiar with the term "glare" in this context? If not, we may want to use language more familiar to the reader.

Reviewer 2: Maybe: 
```suggestion
 - Calls: Implemented glare free negotiation. This fix prevents potential negotiation problems when two clients try to connect simultaneously.
```

>  - Calls: Implemented a glare free negotiation.

Reviewer 2: ```suggestion
 - Calls: Implemented glare free negotiation.
```

 - The Help link now is not converted to lowercase.
 - Added minor performance improvements on sending a message.
 - Archived channels show a more detailed warning.

### Bug Fixes
 - Calls: Fixed a rare case where the Join Call banner showed that a call started "53 years ago".

>  - Calls: fixed a crash on joining calls.

Reviewer 2: ```suggestion
 - Calls: Fixed a crash on joining calls.
```

- [2.1.0 Release](#id1)
- [2.0.1 Release](#id6)
- [2.0.0 Release](#id11)
- [1.55.1 Release](#id14)
- [1.55.0 Release](#id18)

## 2.2.0 Release
- Release Date: March 16, 2023
- Server Versions Supported: Server v7.1.0+ is required. Self-Signed SSL Certificates are not supported unless the user installs the CA certificate on their device.

### Compatibility
 - **Upgrade to server version v7.1.0 or later is required.** Support for server [Extended Support Release](https://docs.mattermost.com/upgrade/extended-support-release.html) (ESR) 6.3.0 has ended and upgrading to server ESR v7.1.0 or later is required. As we innovate and offer newer versions of our mobile apps, we maintain backwards compatibility only with supported server versions. Users who upgrade to the newest mobile apps while being connected to an unsupported server version can be exposed to compatibility issues, which can cause crashes or severe bugs that break core functionality of the app.
 - Android operating system 7+ [is required by Google](https://android-developers.googleblog.com/2017/12/improving-app-security-and-performance.html).	
 - iPhone 5s devices and later with iOS 12.1+ is required.

### Improvements

>  - Added the ability to set the mobile notifications preferences per channel.

Reviewer: ```suggestion
 - Added the ability to set mobile notifications preferences per channel.
```

 - Added support for connecting the WebSocket over TLS1.3.
 - Calls: Added slash commands to start/stop call recordings (``/call recording [start|stop]``).
 - Calls: Implemented glare free negotiation. This fix prevents potential negotiation problems when two clients try to connect simultaneously.
 - The Help link now is not converted to lowercase.
 - Added minor performance improvements on sending a message.
 - Archived channels show a more detailed warning.

### Bug Fixes
 - Calls: Fixed a rare case where the Join Call banner showed that a call started "53 years ago".
 - Calls: Fixed a crash on joining calls.

>  - Fixed an issue where tapping "Send Message" in a user’s profile pop-over did not open a Direct Message channel.

Reviewer: ```suggestion
 - Fixed an issue where tapping **Send Message** in a user’s profile pop-over did not open a Direct Message channel.
```

 - Fixed an issue where an incorrect skin tone was applied to emojis selected in the emoji picker.

>  - Fixed an issue where the channel list displayed Direct Messages with users that had been deactivated.

Reviewer: ```suggestion
 - Fixed an issue where the channel list displayed Direct Messages with user accounts that had been deactivated.
```

>  - Fixed an issue with searching for channels and users with non-Latin characters.

Reviewer: ```suggestion
 - Fixed an issue with searching for channels and users that contain non-Latin characters.
```

 - Fixed an issue where selecting an item from the autocomplete doubled tilde and slash characters.
 - Fixed an "Unable to reset your password" issue.
 - Fixed an issue where the Group Message member count showed as 0 on GraphQL enabled instances.

>  - Fixed an issue with missing posts in a thread when a post got deleted.

Reviewer: ```suggestion
 - Fixed an issue with missing posts in a thread when a post gets deleted.
```
